File: content/code-examples/tested/csharp/driver/Aggregation/Builders/GeoNear.snippet.geo-near.cs
var pipeline = new EmptyPipelineDefinition<Theater>()
    .GeoNear(
        GeoJson.Point(GeoJson.Geographic(-74.1, 40.95)),
        new GeoNearOptions<Theater, Theater>
        {
            DistanceField = "distance",
            MaxDistance = 8000,
            Key = "location.geo",
            Query = Builders<Theater>.Filter.Eq(
                t => t.Location.Address.State, "NJ"),
        });

File: content/code-examples/tested/command-line/mongosh/geospatial/query/geo-near.snippet.geo-near.js
db.theaters.aggregate( [
   {
      $geoNear: {
         near: { type: "Point", coordinates: [ -73.9667, 40.78 ] },
         spherical: true,
         query: { "location.address.state": "NY" },
         distanceField: "calcDistance"
      }
   },
   { $limit: 3 }
] )
